### DISCUSSION TITLE
Slow queries : what can I do ?

### DISCUSSION THREAD

**Discussion by valentinChantelot** [strapi/strapi#8715] [Legacy Topics]
Hi there, 

I'm currently experiencing an issue making my app unusable: very slow queries.

I have to retrieve about 30 entities, composed of simple components and pictures, but the requests are unusually long, around 30s.

I searched quite a bit for a solution, tried to better target the data to be recovered in the controller, and finally reduced the time by almost 2. But 14s is far too long for a correct user experience, especially considering the amount of data to get back.

I saw that several people encounter the same problem as me, and that it would be related to auto populate.

I have very few backend and databases knowledge and this is the first time I created a backend. (that's why I use Strapi ^^') 

Do you have any solution for that ? 

Thanks, 

Valentin

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io/). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

**Reply by valentinChantelot** [reply]
Hi, 

Ok, I'll check the guideline and ask for help in the forum. 

Thanks !

**Comment by mijorus** [comment]
14s? looks like something is wrong. it should take less than one second. does your server have enough bandwidth?

**Reply by valentinChantelot** [reply]
Hi @mijorus, 

Thanks for your answer. 

Everything is fine on server side, my bandwidth is good, and I have enough power and memory to process operations.

As suggested by @derrickmehaffy , I opened a topic on Strapi forum. So, if devs faces the same issue, they can have access to a solution (which is not identified yet) in the future. 

Feel free to join here : https://forum.strapi.io/t/slow-queries-what-can-i-do/1295 😁
